@xen-orchestra/web-core/lib/components/counter/VtsCounter.vue

```diff
@@ -0,0 +1,147 @@
+
+<script lang="ts" setup>
+import { toVariants } from '@core/utils/to-variants.util'
+import { computed } from 'vue'
+
+type CounterAccent = 'brand' | 'neutral' | 'success' | 'warning' | 'danger' | 'muted'
+type CounterVariant = 'primary' | 'secondary'
+type CounterSize = 'small' | 'medium'
+
+const props = defineProps<{
+  value: number | string
+  accent: CounterAccent
+  variant: CounterVariant
+  size: CounterSize
+}>()
+
+const typoClasses = {
+  small: 'typo p4-semi-bold',
+  medium: 'typo p1-medium',
+}
+
+const classNames = computed(() => {
+  return [
+    typoClasses[props.size],
+    toVariants({
+      accent: props.accent,
+      variant: props.variant,
+      size: props.size,
+    }),
+  ]
+})
+</script>
+
+<style lang="postcss" scoped>
+/*
+VARIANT + ACCENT
+--vts-counter--background-color
+--vts-counter--color
+*/
+.vts-counter {
+  &.variant--primary {
+    &.accent--brand {
+      --vts-counter--background-color: var(--color-normal-item-base);
+      --vts-counter--color: var(--color-normal-txt-item);
+    }
+
+    &.accent--neutral {
+      --vts-counter--background-color: var(--color-neutral-txt-primary);
+      --vts-counter--color: var(--color-neutral-background-primary);
+    }
+
+    &.accent--success {
+      --vts-counter--background-color: var(--color-success-item-base);
+      --vts-counter--color: var(--color-success-txt-item);
+    }
+
+    &.accent--warning {
+      --vts-counter--background-color: var(--color-warning-item-base);
+      --vts-counter--color: var(--color-warning-txt-item);
+    }
+
+    &.accent--danger {
+      --vts-counter--background-color: var(--color-danger-item-base);
+      --vts-counter--color: var(--color-danger-txt-item);
+    }
+
+    &.accent--muted {
+      --vts-counter--background-color: var(--color-neutral-background-disabled);
+      --vts-counter--color: var(--color-neutral-txt-secondary);
+    }
+  }
+
+  &.variant--secondary {
+    &.accent--brand {
+      --vts-counter--background-color: var(--color-normal-background-selected);
+      --vts-counter--color: var(--color-normal-txt-base);
+    }
+
+    &.accent--neutral {
+      --vts-counter--background-color: var(--color-neutral-background-secondary);
+      --vts-counter--color: var(--color-neutral-txt-secondary);
+    }
+
+    &.accent--success {
+      --vts-counter--background-color: var(--color-success-background-selected);
+      --vts-counter--color: var(--color-success-txt-base);
+    }
+
+    &.accent--warning {
+      --vts-counter--background-color: var(--color-warning-background-selected);
+      --vts-counter--color: var(--color-warning-txt-base);
+    }
+
+    &.accent--danger {
+      --vts-counter--background-color: var(--color-danger-background-selected);
+      --vts-counter--color: var(--color-danger-txt-base);
+    }
+
+    &.accent--muted {
+      --vts-counter--background-color: var(--color-neutral-background-secondary);
+      --vts-counter--color: var(--color-neutral-txt-secondary);
+    }
+  }
```

Comment:
(same suggestion for all other variants + accent)
To avoid multiple repetitions, what do you think of: 
```postcss
&.variant--secondary {
  --vts-counter--background-color: var(--color-normal-background-selected);
  --vts-counter--color: var(--color-normal-txt-base);

  &.accent--neutral,
  &.accent--muted {
    --vts-counter--background-color: var(--color-neutral-background-secondary);
    --vts-counter--color: var(--color-neutral-txt-secondary);
  }
}
  ```
  
 Personally, I find this more readable since we only change the accent that differs from a "default value."
We don't need to read through all the code to understand that for the `secondary variant`, all accents use 
```postcss
  --vts-counter--background-color: var(--color-normal-background-selected);
  --vts-counter--color: var(--color-normal-txt-base);
```
 except for `neutral` and `muted`.

Reply:
I see the point, and I agree on avoiding repetitions, but in this case, all accents are using different variables:
- for `brand` it is `var(--color-normal-txt-base)`
- for `success` it is `var(--color-success-txt-base)`

and the same for the background color and the other accents.

Or maybe I misinterpreted your comment?

Reply:
Ho MB! I misread the name of the variables used for the colors